### PR TITLE
compiler: use `OnceCell` from std

### LINKED ISSUE #76192: Replace `lazy_static` and `once_cell` crate in rustc crates with std::sync::lazy
Follow up to #75700 

cc @marmeladema

### PR DESCRIPTION
**Pull request by marmeladema**
Fixes #76192 

The only remaining direct use of `lazy_static` crate is in `src/bootstrap`  but I am not sure how I can remove that dependency for now.

r? @matklad

### CODE REVIEW

File: compiler/rustc_interface/src/util.rs [hunk 1]
```diff
@@ -25,6 +25,7 @@ use rustc_span::symbol::{sym, Symbol};
 use smallvec::SmallVec;
 use std::env;
 use std::io::{self, Write};
+use std::lazy::SyncOnceCell as OnceCell;
```
Comment (matklad) [inline compiler/rustc_interface/src/util.rs:28]: Let's avoid the rename here

File: compiler/rustc_interface/src/passes.rs [hunk 2]
```diff
@@ -46,6 +45,7 @@ use std::any::Any;
 use std::cell::RefCell;
 use std::ffi::OsString;
 use std::io::{self, BufWriter, Write};
+use std::lazy::SyncLazy as Lazy;
```
Comment (matklad) [inline compiler/rustc_interface/src/passes.rs:48]: Let's avoid the rename here as well, and fix call-sites to use `SyncLazy`

File: compiler/rustc_interface/src/util.rs (hunk 1 above)
Comment (marmeladema) [reply compiler/rustc_interface/src/util.rs]: Fixed!

File: compiler/rustc_interface/src/passes.rs (hunk 2 above)
Comment (marmeladema) [reply compiler/rustc_interface/src/passes.rs]: Fixed!

**Comment by matklad** [comment]
@bors r+

Thanks!

**Pull request merged**